### PR TITLE
Add formula fizsh

### PR DESCRIPTION
**Pull request by tekacs**
brew has factored out pieces of this (zsh-history-substring-search, for example), but this is a preconfigured, complete fish-like shell for zsh.

Upstream appears to be maintained, being updated somewhat regularly.

### CODE REVIEW

File: Library/Formula/fizsh.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    assert_equal "fizsh, version #{version}", `#{bin}/fizsh -v`.strip
```
Comment (mikemcquaid) [inline Library/Formula/fizsh.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/fizsh.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Fizsh < Formula
+  homepage "https://github.com/zsh-users/fizsh"
+
+  head "https://github.com/zsh-users/fizsh", :using => :git
+
+  stable do
+    url "https://downloads.sourceforge.net/project/fizsh/fizsh-1.0.8.tar.gz"
+    sha1 "515f8828c8bd9f2da1e2716bb3d60727e2f26e90"
+  end
```
Comment (mikemcquaid) [inline Library/Formula/fizsh.rb]: Just a cosmetic change but can you move this above `head`? Thanks!

**Comment by tekacs** [comment]
Changes made. How's that?

Best functional tests I can do without using something like 'expect', since fizsh reverts back to zsh on most-every flag.

File: Library/Formula/fizsh.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    assert_equal "fizsh, version #{version}", `#{bin}/fizsh -v`.strip
+    assert_equal "hello", shell_output("#{bin}/fizsh -c \"echo hello\"").strip
+    assert_equal "zsh", shell_output("#{bin}/fizsh -c \"echo $SHELL\"").strip.split('/')[-1]
```
Comment (mikemcquaid) [inline Library/Formula/fizsh.rb]: Can just do one of these three (let's just go for the "hello" one). Yeh, nice work here these are what I was looking for :+1:

**Comment by tekacs** [comment]
This is updated according to your request, as of a day or so ago. Back to you whenever. :)

**Comment by tekacs** [comment]
Bump. Just since this seems to meet the requests and can be automerged. :)

**Pull request closed without merging**